lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -85,82 +91,369 @@ describe('NxTable', function() {
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow isFilterHeader={null}>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+    });
+  });
+
+  describe('NxTableCell', function() {
+    const renderHeaderComponent = (extraProps?: NxTableCellProps) => render(
+      <NxTable>
+        <NxTableHead>
+          <NxTableRow>
+            <HeaderContext.Provider value={true}>
+              <NxTableCell { ...extraProps }/>
+            </HeaderContext.Provider>
+          </NxTableRow>
+        </NxTableHead>
+      </NxTable>
+    ).getByRole('columnheader') as HTMLElement;
+
+    describe('when the chevron prop is true', function() {
+      describe('when not isHeader', function() {
+        it('sets the button accessible name from the RowContext', function() {
+          render(
+            <NxTable>
+              <NxTableBody>
+                <NxTableRow>
+                  <RowContext.Provider value={{ label: 'foobar', isFilterHeader: false }}>
+                    <NxTableCell chevron/>
+                  </RowContext.Provider>
+                </NxTableRow>
+              </NxTableBody>
+            </NxTable>
+          );
+
+          expect(screen.getByRole('button')).toHaveAccessibleName('foobar');
+        });
+      });
+
+      describe('when isHeader', function() {
+        it('renders a visually hidden header cell with the text "Select Row"', function() {
+          const component = renderHeaderComponent({
+            chevron: true,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          })!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
       });
     });
-    component.update();
 
-    expect(component.find('td')).toHaveProp('colSpan', 1);
-    expect(component.find('td')).toHaveClassName('nx-cell--meta-info');
-    expect(component.find('td')).toIncludeText('Error!');
+    describe('when the rowBtnIcon prop is set', function() {
```

Comment:
Is this describe name wrong? It seems like all of these tests are about `isSortable`.

Reply:
> > Is this describe name wrong? It seems like all of these tests are about isSortable.
> 
> But at the same time, I don't see tests for `rowBtnIcon`.

Cleaned that up and added rowBtnIcon tests here: 7ec076697da54552d9fc99478cb00360d11bf7b9